### PR TITLE
DOC: add links to pre-commit in contributing

### PR DESCRIPTION
**Pull request by feefladder**
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry
The contributing page has comments on style, but does not link to setting up pre-commits yourself. It would be nice to be guided there

### CODE REVIEW

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Full stop is missing. Can you also post a screenshot to show that this renders correctly?

**Comment by Joeperdefloep** [comment]
Now I remember, the main why this is necessary, is that this
> Ensure all linting tests pass, see here for how to run them

links to this page, without any further link to how to set up pre-commits.
here you go:
docs
The period is fixed also.
The build process takes very long on my machine (multiple minutes). Is there a way to make it go faster?
Also, I got the following errors while building, unrelated to what I did:
<details>
```
WARNING: 
>>>-------------------------------------------------------------------------
Exception in /mnt/data/Git/pd_backup/doc/source/whatsnew/v0.9.1.rst at block ending on line 81
Specify :okexcept: as an option in the ipython:: block to suppress this message
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-6-b548a30a7bce> in <module>
----> 1 df = DataFrame(np.random.randn(5, 3), columns = ['A','B','C'])

NameError: name 'DataFrame' is not defined

<<<-------------------------------------------------------------------------
/mnt/data/Git/pd_backup/doc/source/whatsnew/v1.3.0.rst:1250: WARNING: Cannot find contributors for range 'v1.2.5..v1.3.0': Cmd('git') failed due to: exit code(128)
  cmdline: git log --grep=Co-authored --pretty=%b v1.2.5..v1.3.0
  stderr: 'fatal: ambiguous argument 'v1.2.5..v1.3.0': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]''
/mnt/data/Git/pd_backup/doc/source/whatsnew/v1.3.1.rst:65: WARNING: Cannot find contributors for range 'v1.3.0..v1.3.1': Cmd('git') failed due to: exit code(128)
  cmdline: git log --grep=Co-authored --pretty=%b v1.3.0..v1.3.1
  stderr: 'fatal: ambiguous argument 'v1.3.0..v1.3.1': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]''
/mnt/data/Git/pd_backup/doc/source/whatsnew/v1.3.2.rst:65: WARNING: Cannot find contributors for range 'v1.3.1..v1.3.2': Cmd('git') failed due to: exit code(128)
  cmdline: git log --grep=Co-authored --pretty=%b v1.3.1..v1.3.2
  stderr: 'fatal: ambiguous argument 'v1.3.1..v1.3.2': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]''
/mnt/data/Git/pd_backup/doc/source/whatsnew/v1.3.3.rst:78: WARNING: Cannot find contributors for range 'v1.3.2..v1.3.3|HEAD': Cmd('git') failed due to: exit code(128)
  cmdline: git log --grep=Co-authored --pretty=%b v1.3.2..HEAD
  stderr: 'fatal: ambiguous argument 'v1.3.2..HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]''
WARNING: 
>>>-------------------------------------------------------------------------
Exception in /mnt/data/Git/pd_backup/doc/source/whatsnew/v0.18.0.rst at block ending on line 1006
Specify :okexcept: as an option in the ipython:: block to suppress this message
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-108-c90c8ac4f6a1> in <module>
----> 1 s.between_time("7:00am", "9:00am")

/mnt/data/Git/pd_backup/pandas/core/generic.py in between_time(self, start_time, end_time, include_start, include_end, axis)
   7681             raise TypeError("Index must be DatetimeIndex")
   7682 
-> 7683         indexer = index.indexer_between_time(
   7684             start_time, end_time, 
```
</details>

**Comment by MarcoGorelli** [comment]
thanks for spotting this - indeed, since the docs were reorganised, "code standards" is in a different page.

Could you also change the pull request template to link to https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit then please?

> The build process takes very long on my machine (multiple minutes). Is there a way to make it go faster?
Also, I got the following errors while building, unrelated to what I did:

You can just build the doc page you've edited using `--single` (this should be somewhere in the contributing guide)

**Comment by Joeperdefloep** [comment]
Aah, yes, I was initially thinking of elaborating in the contributing page, which was what this was for. Now this does both.

the `--single` cannot find the file, I guess that needs more fumbling around ;)

File: doc/source/development/contributing.rst [hunk 1]
```diff
@@ -331,7 +331,12 @@ can comment::
 
     @github-actions pre-commit
 
-on that pull request. This will trigger a workflow which will autofix formatting errors.
+on that pull request. This will trigger a workflow which will autofix formatting
+errors.
+
+To automatically fix formatting errors on each commit you make, you can
+set up pre-commit yourself. First, create a Python :ref:`environment
+<contributing.python-env>` and then set up :ref:`pre-commit <contributing.pre-commit>`.
```
Comment (MarcoGorelli) [inline doc/source/development/contributing.rst:339]: We can run `pre-commit` in a Docker container, right? If so, I think the first link should be to the https://pandas.pydata.org/pandas-docs/dev/development/contributing_environment.html page

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
> the --single cannot find the file, I guess that needs more fumbling around ;)

can you show the command you ran?

File: doc/source/development/contributing.rst (hunk 1 above)
Comment (Joeperdefloep) [reply doc/source/development/contributing.rst:339]: the command:
```
python make.py --single source/development/contributing.rst
```

Comment (Joeperdefloep) [reply doc/source/development/contributing.rst:339]: also with "quotes"

Comment (Joeperdefloep) [reply doc/source/development/contributing.rst:339]: gives this output:
```python
Traceback (most recent call last):
  File "make.py", line 377, in <module>
    sys.exit(main())
  File "make.py", line 365, in main
    builder = DocBuilder(
  File "make.py", line 56, in __init__
    single_doc = self._process_single_doc(single_doc)
  File "make.py", line 83, in _process_single_doc
    raise FileNotFoundError(f"File {single_doc} not found")
FileNotFoundError: File source/development/contributing.rst not found
```

Comment (Joeperdefloep) [reply doc/source/development/contributing.rst:339]: but `Path("source/development/contributing.rst").is_file()` inside a terminal returns `True`....

Comment (Joeperdefloep) [reply doc/source/development/contributing.rst:339]: o wow, it joined the `SOURCE_PATH` to it in make.py...

Comment (MarcoGorelli) [reply doc/source/development/contributing.rst:339]: you don't need `source/` - check the example, it has
```
python make.py --single development/contributing.rst
```

**Comment by MarcoGorelli** [comment]
Can you address https://github.com/pandas-dev/pandas/pull/43441/files#r703676173 please?

**Comment by Joeperdefloep** [comment]
Sorry, I had solved it locally but I didn't know that not all branches get pushed when I `git push` on another branch.

**Review by MarcoGorelli — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Joeperdefloep

**Pull request merged**